src/k2/module/k23si/Module.cpp

```diff
@@ -478,21 +478,25 @@ bool K23SIPartitionModule::_isScanDone(const Indexer::Iterator& iter, const dto:
         return true;
     } else if (response_size == _config.paginationLimit()) {
         return true;
+    } else if (_config.scanLimit() > 0 && num_scans == _config.scanLimit()) {
+        return true;
     }
 
     return false;
 }
 
 // Helper for handleQuery. Returns continuation token (aka response.nextToScan)
 dto::Key K23SIPartitionModule::_getContinuationToken(const Indexer::Iterator& iter,
-                    const dto::K23SIQueryRequest& request, dto::K23SIQueryResponse& response, size_t response_size) {
+                    const dto::K23SIQueryRequest& request, dto::K23SIQueryResponse& response,
+                    size_t response_size, uint64_t num_scans) {
     auto ikey = iter.getKey();
     // NB: In all checks below, iter.empty signifies that we're past the last
     // key available from the iterator.
     // Three cases where scan is for sure done:
     // 1. Record limit is reached
     // 2. Iterator is not end() but is >= user endKey
     // 3. Iterator is at end() and partition bounds contains endKey
+    (void)num_scans;
```

Comment:
Can we remove the num_scans parameter from this function instead?

Reply:
Yes, I though I will need it. But seems like I don't

Reply:
Fixed